### PR TITLE
feat: offer package search suggestions

### PR DESCRIPTION
**Pull request by davidmundelius**
### Summary of changes

- Use react query to load search suggestions from deps.dev
- Create faker module for mocking package search hooks

### Context and reason for change

A step toward #2394 

### How can the changes be tested

Start the app and open the file opossum_input.opossum from the example-files folder.
In Audit View, choose the root folder. In the Network tab of the DevTools, check that there is an entry with name "suggest?q=Vue". Check that the respone for this entry contains the desired search results.

### CODE REVIEW

**Review by mstykow — CHANGES_REQUESTED** [review]

File: src/Frontend/util/package-search-hooks.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+// SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+//
+// SPDX-License-Identifier: Apache-2.0
+import { useQuery } from '@tanstack/react-query';
+
+import { PackageSearchApi } from './package-search-api';
+
+function usePackageSearchSuggestions(searchTerm: string | undefined) {
+  const { data, error, isLoading } = useQuery({
+    queryKey: ['search-suggestions'],
```
[the remark this answers is not in the document]
Comment (davidmundelius) [reply src/Frontend/util/package-search-hooks.ts]: Done.

File: src/Frontend/util/package-search-hooks.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    queryFn: () =>
+      searchTerm ? PackageSearchApi.getSuggestions(searchTerm) : {},
+    refetchOnWindowFocus: false,
```
[the remark this answers is not in the document]
Comment (davidmundelius) [reply src/Frontend/util/package-search-hooks.ts]: Done.

File: package.json [hunk 3]
```diff
@@ -51,6 +51,7 @@
     "@faker-js/faker": "^8.3.1",
     "@playwright/test": "^1.40.1",
     "@redux-devtools/extension": "^3.3.0",
+    "@tanstack/eslint-plugin-query": "^5.12.1",
```
Comment (mstykow) [inline package.json:54]: please add this dependency to `renovate.json5` so that it can be updated automatically

**Review by mstykow — CHANGES_REQUESTED** [review]

File: package.json (hunk 3 above)
Comment (mstykow) [reply package.json:54]: ideally you create a `@tanstack` block similar to how this was done for `@emotion` for example

File: renovate.json5 [hunk 4]
```diff
@@ -84,6 +84,11 @@
       groupName: 'Emotion dependencies',
       groupSlug: 'emotion',
     },
+    {
+      matchPackagePrefixes: ['@tanstack/'],
```
Comment (mstykow) [inline renovate.json5:88]: you should now remove the`@tanstack/react-query` dependency from the first block in this file.

File: package.json (hunk 3 above)
Comment (davidmundelius) [reply package.json:54]: Done.

**Review by mstykow — CHANGES_REQUESTED** [review]

File: renovate.json5 (hunk 4 above)
Comment (davidmundelius) [reply renovate.json5:87]: Done.

**Review by mstykow — APPROVED** [review]

**Review by mstykow — CHANGES_REQUESTED** [review]
Let's wait to merge this until the suggestions feature is complete and usable.

**Pull request merged**